src/editor/editdraw.c

```diff
@@ -95,45 +95,61 @@ printwstr (const char *s, int len)
 
 /* --------------------------------------------------------------------------------------------- */
 
-static inline void
-status_string (WEdit *edit, char *s, int w)
+static char *
+format_character_code (WEdit *edit)
 {
-    char byte_str[16];
+    const int bufsize = 16;
```

Comment:
```suggestion
    enum { bufsize = 16 };
```

Not awesome, but probably better than `#define`?

Reply:
I guess the problem is that my code has a VLA?

I really don't like this enum hack. How about BUF_TINY (64)?

Reply:
I didn't like that the buffer size was repeated two times.

Unfortunately, `const int` is a run-time constant, so you can't use it as a fixed array size; you have to use a macro substitution (like you did now, picking the pre-defined `TINY` buffer size) - or go for VLA arrays, which I think is a GNU extension. Or else, if you want to stay within the type system level, use an enum.

What I didn't like with the `#define` option is that it can be changed at one place and forgotten at another. They are usually used for defining fixed-size arrays, and thus only once - at other places one could then use `sizeof`.

Therefore, I suggested to keep it as you did, but pull the `bufsize` to the compile-time, all while keeping it at the type system level. This way, if the number is changed, it cannot be wrongly not updated at some place.

Thus, I think that the enum option actually is much safer and has more advantages than any other options, except for the possibly confusing way it looks.

If I had to choose between different options, I would rate them like this for the reasons above:

1. Enum option
2. Your first solution
3. The way it looks like now

Reply:
An earlier draft contained a `static char buf[16]` in the callee, returning its address. That way I could use `sizeof`. As discussed in the main issue, you're not in favor of this approach, e.g. it could break we ever go multithreaded. I also understand your preference of using the same pattern across the callbase: callee allocates, caller frees.

Unfortunately I'm afraid there's no solution to this problem that'd perfect address all aspects.

My problem with the `enum` approach is that it's the first time I'm ever in my life seeing `enum` being used like that, I don't think it's standard practice, and it's also against `enum`'s semantics. If we really look in this direction, and if you're not okay with using `BUF_TINY` all throughout (why not?), then another brand new `#define` at the top of this method, and `#undef` at the end, would IMO be preferable to `enum` -- at least in the sense that it's a much more typical pattern.

Alternatively, just remove all those `n`s from the function names and the passed lengths. I mean, it's pretty obvious from the format strings that we're printing roughly 10-ish bytes into a 64-byte buffer. Just allocate the buffer and not be paranoid about overrun?